txpool/txpool.go

```diff
@@ -919,11 +924,16 @@ func (p *TxPool) handlePromoteRequest(req promoteRequest) {
 
 // addGossipTx handles receiving transactions
 // gossiped by the network.
-func (p *TxPool) addGossipTx(obj interface{}, _ peer.ID) {
+func (p *TxPool) addGossipTx(obj interface{}, peerID peer.ID) {
 	if !p.sealing.Load() {
 		return
 	}
 
+	// ignore txs gossiped by the node itself
+	if p.localPeerID == peerID {
+		return
+	}
```

Comment:
You could have merged these two conditions, IMO code is self-explanatory.

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/129/commits/b021bef2181a4f5df9f7b855b0a23c1b9adab193